src/test/java/io/pinecone/PineconeIndexOperationClientTest.java

```diff
@@ -0,0 +1,37 @@
+package io.pinecone;
+
+import org.asynchttpclient.*;
+import org.junit.jupiter.api.Test;
+
+import java.io.IOException;
+import java.util.concurrent.CompletableFuture;
+
+import static org.mockito.Mockito.*;
+
+public class PineconeIndexOperationClientTest {
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void testSuccessfulIndexDeletion() throws IOException {
```

Comment:
A negative test case would be nice! does not have to be as comprehensive but here is the test suite for python in case you want ideas : 
https://github.com/pinecone-io/pinecone-db/blob/main/tests/integration/test_control_plane.py

Reply:
Currently the unit test is not making a call to the database and as a part of integration test, I'll be adding various possible negative tests. Thanks for providing me with the python reference.